packages/legend-query/src/components/QueryBuilderProjectionPanel.tsx

```diff
@@ -280,12 +281,29 @@ const QueryBuilderProjectionColumnEditor = observer(
     const onContextMenuClose = (): void => setIsSelectedFromContextMenu(false);
 
     const { projectionColumnState, isRearrangingColumns } = props;
+    const applicationStore = useApplicationStore();
     const queryBuilderState =
       projectionColumnState.projectionState.queryBuilderState;
     const projectionState =
       queryBuilderState.fetchStructureState.projectionState;
-    const removeColumn = (): void =>
-      projectionState.removeColumn(projectionColumnState);
+    const removeColumn = (): void => {
+      const columnsUsedInPostFilter = Array.from(
+        queryBuilderState.postFilterState.nodes.values(),
```

Comment:
I think we should disable the button so users can't even click on the delete button.